Modules/CalcDefence.lua

```diff
@@ -1113,68 +1116,103 @@ function calcs.defence(env, actor)
-				s_format("Maximum hit you can take: %d", output[damageType.."MaximumHitTaken"]),
+			breakdown[damageType.."MaximumHitTaken"] = { 
+				label = "Maximum Hit Taken (uses lowest value)",
+				rowList = { },
+				colList = {
+					{ label = "Type", key = "type" },
+					{ label = "TotalPool", key = "pool" },
+					{ label = "Taken", key = "taken" },
+					{ label = "Final", key = "final" },
+				},
 			}
 		end
+		output[damageType.."MaximumHitTaken"] = math.huge
+		for _, damageConvertedType in ipairs(dmgTypeList) do
+			if actor.damageShiftTable[damageType][damageConvertedType] > 0 then
+				local hitTaken = output[damageConvertedType.."TotalPool"] / (actor.damageShiftTable[damageType][damageConvertedType] / 100) / output[damageConvertedType.."BaseTakenHitMult"]
+				if hitTaken < output[damageType.."MaximumHitTaken"] then
+					output[damageType.."MaximumHitTaken"] = hitTaken
+				end
+				
+				if breakdown then
+					t_insert(breakdown[damageType.."MaximumHitTaken"].rowList, {
+						type = s_format("%d%% as %s", actor.damageShiftTable[damageType][damageConvertedType], damageConvertedType),
+						pool = s_format("x %d", output[damageConvertedType.."TotalPool"]),
+						taken = s_format("/ %.2f", output[damageConvertedType.."BaseTakenHitMult"]),
+						final = s_format("x %.0f", hitTaken),
+					})
+				end
+			end
+		end
+		if breakdown then
+			 t_insert(breakdown[damageType.."MaximumHitTaken"], s_format("Total Pool: %d", output[damageType.."TotalPool"]))
+			 t_insert(breakdown[damageType.."MaximumHitTaken"], s_format("Taken Mult: %.2f",  output[damageType.."TotalPool"] / output[damageType.."MaximumHitTaken"]))
+			 t_insert(breakdown[damageType.."MaximumHitTaken"], s_format("Maximum hit you can take: %d", output[damageType.."MaximumHitTaken"]))
+		end
 	end
-
+	
+	
 	local DamageTypeConfig = env.configInput.EhpCalcMode or "Average"
-	local minimumEHP = 2147483648
-	local minimumEHPMode = "NONE"
-	if DamageTypeConfig == "Minimum" then
-		DamageTypeConfig = {"Melee", "Projectile", "Spell", "SpellProjectile"}
-		minimumEHPMode = "Melee"
-	else
-		DamageTypeConfig = {DamageTypeConfig}
-	end
-	for _, DamageType in ipairs(DamageTypeConfig) do
-		--total EHP
-		for _, damageType in ipairs(dmgTypeList) do
-			local convertedAvoidance = 0
-			for _, damageConvertedType in ipairs(dmgTypeList) do
-				convertedAvoidance = convertedAvoidance + output[damageConvertedType..DamageType.."DamageChance"] * actor.damageShiftTable[damageType][damageConvertedType] / 100
-			end
-			output[damageType.."TotalEHP"] = output[damageType.."MaximumHitTaken"] / (1 - output[DamageType.."NotHitChance"] / 100) / (1 - convertedAvoidance / 100)
-			if minimumEHPMode ~= "NONE" then
-				if output[damageType.."TotalEHP"] < minimumEHP then
-					minimumEHP = output[damageType.."TotalEHP"]
-					minimumEHPMode = DamageType
+	for _, damageType in ipairs(dmgTypeList) do
+		local DamageType = DamageTypeConfig
```

Comment:
`DamageTypeConfig`. `DamageType`, `DamageTypeList` and `ChanceToTakeDamage` should be in camelCase.